### PR TITLE
Add SonarCloud configs

### PR DESCRIPTION
**Pull request by sanchariGr**
**Proposed changes**:
Screenshot 2023-04-28 at 15 09 32
[Sonarcloud repo](https://sonarcloud.io/component_measures?id=RasaHQ_rasa&pullRequest=12336&metric=new_duplicated_lines_density&view=list)

**Quality gates are set at** : (open to more suggestions)

Screenshot 2023-04-28 at 15 12 44

**Pricing :** 
- This oss integration is currently free
- For private projects [this](https://www.sonarsource.com/plans-and-pricing/#sonarcloud) is how their pricing is calculate. cc @m-vdb 

**Security**
Sonar also does [security](https://sonarcloud.io/project/security_hotspots?id=RasaHQ_rasa) assessments. Also important to note from @rasa-jmac `sonar results are also public for open source repos so if it finds security issues, it also will be widely visible`

### CODE REVIEW

File: .github/workflows/continous-integration.yml [hunk 1]
```diff
@@ -540,15 +540,22 @@ jobs:
...
         env:
-          CC_TEST_REPORTER_ID: ${{ secrets.CODECLIMATE_REPORTER_ID }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
         with:
-          coverageLocations: |
-            ${{ github.workspace }}/coverage.xml:coverage.py
-          debug: true
+          args: >
+            -Dsonar.projectKey=RasaHQ_rasa
```
Comment (radovanZRasa) [inline .github/workflows/continous-integration.yml:555]: Seems like this is a duplicate of what is in `sonar-project.properties`

File: .github/workflows/continous-integration.yml [hunk 2]
```diff
@@ -540,15 +540,22 @@ jobs:
...
+            -Dsonar.python.coverage.reportPaths=coverage.xml
```
Comment (radovanZRasa) [inline .github/workflows/continous-integration.yml:556]: Seems like the rest of these could be moved to `sonar-project.properties`

**Review by radovanZRasa — COMMENTED** [review]
Great job!

We need someone who has admin access to our org on GitHub to go to SonarCloud's app and disable SonarCloud's Automatic analysis since code analysis will be performed during the CI run.
[Disable SonarCloud's Automatic Analysis](https://docs.sonarcloud.io/enriching/test-coverage/python-test-coverage/#use-ci-based-not-automatic-analysis)

**Comment by m-vdb** [comment]
@radovanZRasa looking at SonarCloud doc, you only need to be an admin of SonarCloud to disable the automatic analysis. @sanchariGr I think you can disable it? (I looked at the GH app and there is no such setting)

File: .github/workflows/continous-integration.yml [hunk 3]
```diff
@@ -540,15 +540,22 @@ jobs:
...
           coverage xml
 
-      - name: Upload reports to codeclimate
-        if: needs.changes.outputs.backend == 'true'
-        uses: paambaati/codeclimate-action@b649ad206d2e83dafb9ed130deba698aa1b41d78
+      - name: Analyse code with SonarCloud
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+        with:
+          # Disabling shallow clone is recommended for improving relevancy of reporting
+          fetch-depth: 0
```
Comment (ancalita) [inline .github/workflows/continous-integration.yml:547]: This looks like a duplicate of the first repo checkout step, should we add the `with` over there in L511

File: README.md [hunk 4]
```diff
@@ -6,7 +6,6 @@
 [![PyPI version](https://badge.fury.io/py/rasa.svg)](https://badge.fury.io/py/rasa)
 [![Supported Python Versions](https://img.shields.io/pypi/pyversions/rasa.svg)](https://pypi.python.org/pypi/rasa)
 [![Build Status](https://github.com/RasaHQ/rasa/workflows/Continuous%20Integration/badge.svg)](https://github.com/RasaHQ/rasa/actions)
-[![Coverage Status](https://api.codeclimate.com/v1/badges/756dc6fea1d5d3e127f7/test_coverage)](https://codeclimate.com/github/RasaHQ/rasa/)
```
Comment (ancalita) [inline README.md:9]: Shouldn't we link to the Sonar coverage status here?

**Review by ancalita — COMMENTED** [review]
I have a few questions on metric thresholds - for example, how did we decide on the 3% for duplication and coverage of 80% of new code and 60% overall for the repo (not sure if I read the screenshots correctly)

**Comment by sanchariGr** [comment]
> @radovanZRasa looking at SonarCloud doc, you only need to be an admin of SonarCloud to disable the automatic analysis. @sanchariGr I think you can disable it? (I looked at the GH app and there is no such setting)

Just curious if you can access [this](https://sonarcloud.io/project/analysis_method?id=RasaHQ_rasa) ? I already did turn off the automatic analysis. Will also add @radovanZRasa and @m-vdb as admins so we can play around with the offerings and metrics.

**Comment by sanchariGr** [comment]
> I have a few questions on metric thresholds - for example, how did we decide on the 3% for duplication and coverage of 80% of new code and 60% overall for the repo (not sure if I read the screenshots correctly)

@ancalita good question the metric are still up for discussion. I created a custom metric and based it off sonar's built in suggestions below, just as an example.
Screenshot 2023-05-02 at 11 02 52

**Comment by m-vdb** [comment]
nope, not able to access it

File: .github/workflows/continous-integration.yml (hunk 3 above)
Comment (sanchariGr) [reply .github/workflows/continous-integration.yml]: You're right! Updated.

**Comment by radovanZRasa** [comment]
> 

I am not able to access the page.

**Comment by radovanZRasa** [comment]
> I have a few questions on metric thresholds - for example, how did we decide on the 3% for duplication and coverage of 80% of new code and 60% overall for the repo (not sure if I read the screenshots correctly)

These are the default values in Sonar Cloud.
We should start to track these metrics and see if we should re-adjust them to fit our needs.

**Comment by sanchariGr** [comment]
> nope, not able to access it

@m-vdb I've added you and Rasa as an admin on the sonar org, you should have access to it now.

File: .github/workflows/continous-integration.yml [hunk 5]
```diff
@@ -499,7 +499,7 @@ jobs:
             ${{ github.workspace }}/${{ matrix.test }}-coverage
 
   upload_coverage_reports:
-    name: Upload coverage reports to codeclimate
+    name: Upload coverage reports
```
Comment (radovanZRasa) [inline .github/workflows/continous-integration.yml:502]: The name is not fitting to what this job is doing.

File: .github/workflows/continous-integration.yml [hunk 6]
```diff
@@ -540,15 +540,26 @@ jobs:
           coverage combine "${final_dir}/"*
           coverage xml
 
-      - name: Upload reports to codeclimate
+  sonarcloud:
+    name: SonarCloud Scan
+    if: github.ref_type != 'tag'
+    runs-on: ubuntu-22.04
+    needs:
+      - upload_coverage_reports
```
Comment (vcidst) [inline .github/workflows/continous-integration.yml:548]: Did you forget to update the job name here?

**Review by vcidst — COMMENTED** [review]
Looks great! I've just one comment

File: .github/workflows/continous-integration.yml (hunk 6 above)
Comment (sanchariGr) [reply .github/workflows/continous-integration.yml:548]: Yup i did, thanks:)

**Review by radovanZRasa — APPROVED** [review]

**Review by vcidst — APPROVED** [review]
💯

**Pull request merged**